packages/teams-js/src/public/appWindow.ts

```diff
@@ -44,16 +53,21 @@ export class ChildAppWindow implements IAppWindow {
    * @param type - The event to listen to. Currently the only supported type is 'message'.
    * @param listener - The listener that will be called
    */
-  public addEventListener(type: string, listener: (message: any) => void): void {
+  public addEventListener(type: string, listener: addEventListnerFunctionType): void {
     ensureInitialized(runtime);
     if (type === 'message') {
       registerHandler('messageForParent', listener);
     }
   }
 }
 
+/**
+ * Parent app window object that can be used to send messages to and fro from child window.
```

Comment:
>Parent app window object [](http://example.com/codeflow?start=3&length=24)

Can you add more details about in what context a developer would use this type?

Reply:
Does this sounds better 
```
Developers can use parent app window objects to create the main UI for their app like notifications, chat integration, or access to Host APIs.

Reply:
Is this class actually used anywhere?

Reply:
Looking at the library we did change this:
``` 
In `dialog`, `ParentAppWindow.postMessage` was renamed to `dialog.sendMessageToParent(message: any): void`. `ParentAppWindow.addEventListener` was renamed to `dialog.registerOnMessageFromParent`. 

Will mark this class as deprecated in the latest commit.

Reply:
As Trevor mentioned in his comment this should be something like this:
``` 
This is used for communicating with a parent window that opened your current window (so a dialog/task module would use it to send messages to the application that opened it)